packages/openapi-react-query/src/index.ts

```diff
@@ -141,9 +142,14 @@ export default function createClient<Paths extends {}, Media extends MediaType =
             const mth = method.toUpperCase() as Uppercase<typeof method>;
             const fn = client[mth] as ClientMethod<Paths, typeof method, Media>;
             const { data, error } = await fn(path, init as InitWithUnknowns<typeof init>);
-            if (error || !data) {
+            if (error) {
               throw error;
             }
+
+            if (data === undefined) {
```

Comment:
Since mutationFn type doesn't allow returning undefined I had to resort to this - this changes return type a little bit but should be fine since data should only be a valid JSON and undefined is not valid. 

React-query does allow returning undefined from mutationFn tho - that restriction is only for queries. 

But the problem here is at the type level - data is basically `T | undefined` but mutationFn basically has to return `T`. 
- If we don't do an undefined check - we're returning `undefined` which not part of `T`. 
- If do an undefined check - what if `T` was instantiated as `number | undefined`. Then we're excluding `undefined` from the return type even if it part of `T` type. 

We don't know at runtime if data is undefined because it `fn` errored or because that's what data is - but since it should be impossible to return undefined that's why I went with the 2nd option

Reply:
`data` is `T | undefined` but the return type says we can only return `T` - so we somehow need to exclude undefined from `data` even though `T` itself can containe undefined. It can be tricky to do so - I've assumed before that T can only be valid JSON, thus can't in practice be undefined but it seems that's not true.